### PR TITLE
Handling of expected error codes coming from grpc storage plugins #1741

### PR DESCRIPTION
**Pull request by chandresh-pancholi**
Signed-off-by: chandresh-pancholi <chandreshpancholi007@gmail.com>

## Which problem is this PR solving?
- #1741 

## Short description of the changes
- Changing the error code

### CODE REVIEW

File: plugin/storage/grpc/shared/grpc_client.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func (c *grpcClient) GetTrace(ctx context.Context, traceID model.TraceID) (*mode
 	trace := model.Trace{}
 	for received, err := stream.Recv(); err != io.EOF; received, err = stream.Recv() {
 		if err != nil {
-			return nil, errors.Wrap(err, "grpc stream error")
+			return nil, spanstore.ErrTraceNotFound
```
Comment (yurishkuro) [inline plugin/storage/grpc/shared/grpc_client.go]: This doesn't look right, why are you assuming that EVERY error from gRPC is "trace not found"?

Comment (chandresh-pancholi) [reply plugin/storage/grpc/shared/grpc_client.go]: fixed.

File: plugin/storage/grpc/shared/grpc_client_test.go [hunk 2]
```diff
@@ -166,6 +166,20 @@ func TestGRPCClientGetTrace_NoTrace(t *testing.T) {
...
+func TestGRPCClientGetTrace_StreamErrorTraceNotFound(t *testing.T) {
+	withGRPCClient(func(r *grpcClientTest) {
+		traceClient := new(grpcMocks.SpanReaderPlugin_GetTraceClient)
+		traceClient.On("Recv").Return(nil, spanstore.ErrTraceNotFound)
+		r.spanReader.On("GetTrace", mock.Anything, &storage_v1.GetTraceRequest{
+			TraceID: mockTraceID,
+		}).Return(traceClient, nil)
+
+		s, err := r.client.GetTrace(context.Background(), mockTraceID)
+		assert.Error(t, err)
```
Comment (yurishkuro) [inline plugin/storage/grpc/shared/grpc_client_test.go]: that doesn't test the issue, you probably want ErrorEquals method

Comment (chandresh-pancholi) [reply plugin/storage/grpc/shared/grpc_client_test.go]: fixed.

**Pull request merged**